File: csharp/matrix/Matrix.cs
public class Matrix
{
    private readonly int[][] _matrix;
    public Matrix(string input)
    {
        _matrix = input
                    .Split('\n')
                    .Select(row => row.Split(' ')
                                    .Select(int.Parse)
                                    .ToArray())
                    .ToArray();
    }
    public int Rows => _matrix.Length;
    public int Cols => _matrix.FirstOrDefault()?.Length ?? 0;

    public int[] Row(int row)
    => _matrix[row - 1];
    public int[] Column(int col)
    => _matrix.Select(row => row[col - 1]).ToArray();
}

File: typescript/matrix/matrix.ts
export class Matrix {
  private _matrix: number[][];

  constructor(matrixString: string) {
    this._matrix = matrixString.split(`\n`).map(row => row.split(` `).map(Number));
  }

  get rows(): number[][] {
    return this._matrix;
  }

  get columns(): number[][] {
    return this._matrix[0].map((_, i) => this._matrix.map(row => row[i]));
  }
}
